hosting/docker-compose.build.yaml

```diff
@@ -0,0 +1,118 @@
+version: "3"
```

Comment:
![aXj57Bg_460s](https://github.com/Budibase/budibase/assets/8755148/b4a32aa8-49f5-44b9-81d0-b295acb5c844)

Reply:
Rather than having another locally running:
- `minio-service`
- `proxy-service`
- `couchdb-service`

I would greatly prefer to be able to run the dockerised server / worker alongside my existing infrastructure in `docker-compose.dev.yaml`. 

Often it'll be the case that you'll want to test something that you've been testing in dev and already have data setup for. If we could:

- `docker-compose.infra.yaml` -> replacement for `dev`, name change only
- `docker-compose.services.yaml` -> this file renamed, remove infra, keep server / worker

Then it'd be simpler to be able to 
- Run services via ide during development
- Stop those
- Run the docker services instead
- Pickup where I left off 


Cc: @mike12345567 @shogunpurple for thoughts here.

Reply:
@mike12345567 you built this one are you are closer to the different configurations. Do you mind having a look at this?